src/app/files.js

```diff
@@ -107,6 +107,47 @@ function Files (storage) {
     return files
   }
 
+  //
+  // Tree model for files
+  // {
+  //   'a': { }, // empty directory 'a'
+  //   'b': {
+  //     'c': {}, // empty directory 'b/c'
+  //     'd': { '/readonly': true, '/content': 'Hello World' } // files 'b/c/d'
+  //     'e': { '/readonly': false, '/path': 'b/c/d' } // symlink to 'b/c/d'
```

Comment:
```js
var fs1 = {
  b: {
    '/readonly': '...'
    c: {
      '/readonly': '...'
      // c is an empty directory?
    }
  }
}

var fs2 = {
  b: {
    '/readonly': '...'
    c: {
      '/readonly': '...'
      '/content': 'c is a file'
    }
  }
}

var fs2 = {
  b: {
    '/readonly': '...'
    c: {
      '/readonly': '...'
      '/path': 'a'
    }
  },
  a: {
    // empty directory
  }
}
```

* when updating the structure, it's important to keep `/readonly` and `/mode` in sync.
* when updating the structure, it's important to take care, that a node does not have `/content` and `/path`
* if the `/path` or `/content` is set, then a node can still have "subdirectories", but they are hidden, because the object is not interpreted as a directory.
* iterating over a node in case it's a directory requires filtering out all properties that start with `/`